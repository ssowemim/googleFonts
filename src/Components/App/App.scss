.App {
  .App__Btn {
    appearance: none;
    background: #ffdf01;
    border-radius: 4px;
    border: none;
    cursor: pointer;
    font-size: 13px;
    padding: 6px 10px;
    width: 200px;
    position: fixed;
    bottom: 20px;
    right: 20px;

    &:hover {
      background: #000;
      color: #fff;
    }
  }
}

.Wrapper {
  max-width: 800px;
  width: 100%;
  margin: 0 auto;
  display: flex;
  flex-direction: column;
}

.TexBoxContainer {
  margin-top: 20px;
}

.App-intro {
  display: none
}

.HideControl {
  .Header .App__header {
    left: -160px
  }

  .TextBox .TextBox__CTA {
    display: none;
  }

  .TextBox.isActive {
    background-color: transparent;
  }

  .FontOptionContainer {
    right: -250px;
  }
}
