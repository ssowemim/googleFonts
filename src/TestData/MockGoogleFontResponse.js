export const googleFontResponse = {
    kind: "webfonts#webfontList",
    items: [
        {
            kind: "webfonts#webfont",
            family: "ABeeZee",
            category: "sans-serif",
            variants: [
                "regular",
                "italic"
            ],
            subsets: [
                "latin"
            ],
            version: "v11",
            lastModified: "2017-10-10",
            files: {
                regular: "http://fonts.gstatic.com/s/abeezee/v11/mE5BOuZKGln_Ex0uYKpIaw.ttf",
                italic: "http://fonts.gstatic.com/s/abeezee/v11/kpplLynmYgP0YtlJA3atRw.ttf"
            }
        },
        {
            kind: "webfonts#webfont",
            family: "Abel",
            category: "sans-serif",
            variants: [
                "regular"
            ],
            subsets: [
                "latin"
            ],
            version: "v8",
            lastModified: "2017-10-10",
            files: {
                regular: "http://fonts.gstatic.com/s/abel/v8/RpUKfqNxoyNe_ka23bzQ2A.ttf"
            }
        },
        {
            kind: "webfonts#webfont",
            family: "Abhaya Libre",
            category: "serif",
            variants: [
                "regular",
                "500",
                "600",
                "700",
                "800"
            ],
            subsets: [
                "latin",
                "latin-ext",
                "sinhala"
            ],
            version: "v3",
            lastModified: "2017-10-10",
            files: {
                500: "http://fonts.gstatic.com/s/abhayalibre/v3/wBjdF6T34NCo7wQYXgzrc5MQuUSAwdHsY8ov_6tk1oA.ttf",
                600: "http://fonts.gstatic.com/s/abhayalibre/v3/wBjdF6T34NCo7wQYXgzrc2v8CylhIUtwUiYO7Z2wXbE.ttf",
                700: "http://fonts.gstatic.com/s/abhayalibre/v3/wBjdF6T34NCo7wQYXgzrc0D2ttfZwueP-QU272T9-k4.ttf",
                800: "http://fonts.gstatic.com/s/abhayalibre/v3/wBjdF6T34NCo7wQYXgzrc_qsay_1ZmRGmC8pVRdIfAg.ttf",
                regular: "http://fonts.gstatic.com/s/abhayalibre/v3/zTLc5Jxv6yvb1nHyqBasVy3USBnSvpkopQaUR-2r7iU.ttf"
            }
        },
        {
            kind: "webfonts#webfont",
            family: "Abril Fatface",
            category: "display",
            variants: [
                "regular"
            ],
            subsets: [
                "latin",
                "latin-ext"
            ],
            version: "v9",
            lastModified: "2017-10-10",
            files: {
                regular: "http://fonts.gstatic.com/s/abrilfatface/v9/X1g_KwGeBV3ajZIXQ9VnDojjx0o0jr6fNXxPgYh_a8Q.ttf"
            }
        },
        {
            kind: "webfonts#webfont",
            family: "Aclonica",
            category: "sans-serif",
            variants: [
                "regular"
            ],
            subsets: [
                "latin"
            ],
            version: "v8",
            lastModified: "2017-10-10",
            files: {
                regular: "http://fonts.gstatic.com/s/aclonica/v8/M6pHZMPwK3DiBSlo3jwAKQ.ttf"
            }
        },
        {
            kind: "webfonts#webfont",
            family: "Acme",
            category: "sans-serif",
            variants: [
                "regular"
            ],
            subsets: [
                "latin"
            ],
            version: "v7",
            lastModified: "2017-10-10",
            files: {
                regular: "http://fonts.gstatic.com/s/acme/v7/-J6XNtAHPZBEbsifCdBt-g.ttf"
            }
        },
        {
            kind: "webfonts#webfont",
            family: "Actor",
            category: "sans-serif",
            variants: [
                "regular"
            ],
            subsets: [
                "latin"
            ],
            version: "v7",
            lastModified: "2017-10-10",
            files: {
                regular: "http://fonts.gstatic.com/s/actor/v7/ugMf40CrRK6Jf6Yz_xNSmQ.ttf"
            }
        },
        {
            kind: "webfonts#webfont",
            family: "Adamina",
            category: "serif",
            variants: [
                "regular"
            ],
            subsets: [
                "latin"
            ],
            version: "v10",
            lastModified: "2017-10-10",
            files: {
                regular: "http://fonts.gstatic.com/s/adamina/v10/RUQfOodOMiVVYqFZcSlT9w.ttf"
            }
        }]
}