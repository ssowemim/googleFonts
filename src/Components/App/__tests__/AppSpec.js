import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import fetchMock from 'fetch-mock';

import App from './../index';
import { googleFontResponse } from './../../../TestData/MockGoogleFontResponse';

describe('App', ()=>{
  xit('renders correctly', () => {
    fetchMock.get('*', googleFontResponse);
    let tree = renderer.create(
      <App />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
})
