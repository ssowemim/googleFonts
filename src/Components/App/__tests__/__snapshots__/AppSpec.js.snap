// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="App"
>
  <div
    className="Header"
  >
    <header
      className="App__header"
    >
      <a
        className="App__Logo"
        href="/"
      >
        <span
          className="App__Logo--First"
        >
          GFonts
        </span>
        <span
          className="App__Logo--Last"
        >
          Space
        </span>
      </a>
      <div
        className="App__SocialIcons"
      >
        <a
          className="App__SocialIcons__Link"
          href="https://www.linkedin.com/in/pankaj-ladhar-51781137/"
          rel="noopener noreferrer"
          target="_blank"
          title="Pankaj Ladhar linkedin profile "
        >
          <i
            className="fa fa-linkedin"
          />
        </a>
        <a
          className="App__SocialIcons__Link"
          href=""
          rel="noopener noreferrer"
          target="_blank"
          title="Pankaj Ladhar github profile "
        >
          <i
            className="fa fa-github"
          />
        </a>
      </div>
    </header>
  </div>
  <section
    className="Wrapper"
  >
    <section
      className="FontOptionContainer"
    >
      <section
        className="FontOptionContainer__Item"
      >
        <h4
          className="FontOptionContainer__Title"
        >
          Classification
        </h4>
        <ul />
      </section>
      <section
        className="FontOptionContainer__Item"
      >
        <h4
          className="FontOptionContainer__Title"
        >
          Propeties
        </h4>
        <div
          className="FontOptionContainer__Properties"
        >
          <div
            className="Select Select--single is-clearable is-searchable"
            style={undefined}
          >
            <div
              className="Select-control"
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              style={undefined}
            >
              <span
                className="Select-multi-value-wrapper"
                id="react-select-2--value"
              >
                <div
                  className="Select-placeholder"
                >
                  Select Font Family
                </div>
                <div
                  className="Select-input"
                  style={
                    Object {
                      "display": "inline-block",
                    }
                  }
                >
                  <input
                    aria-activedescendant="react-select-2--value"
                    aria-describedby={undefined}
                    aria-expanded="false"
                    aria-haspopup="false"
                    aria-label={undefined}
                    aria-labelledby={undefined}
                    aria-owns=""
                    className={undefined}
                    id={undefined}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={false}
                    role="combobox"
                    style={
                      Object {
                        "boxSizing": "content-box",
                        "width": "5px",
                      }
                    }
                    tabIndex={undefined}
                    value=""
                  />
                  <div
                    style={
                      Object {
                        "height": 0,
                        "left": 0,
                        "overflow": "scroll",
                        "position": "absolute",
                        "top": 0,
                        "visibility": "hidden",
                        "whiteSpace": "pre",
                      }
                    }
                  >
                    
                  </div>
                </div>
              </span>
              <span
                className="Select-arrow-zone"
                onMouseDown={[Function]}
              >
                <span
                  className="Select-arrow"
                  onMouseDown={[Function]}
                />
              </span>
            </div>
          </div>
          <div
            className="Select Select--single is-disabled is-searchable"
            style={undefined}
          >
            <div
              className="Select-control"
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              style={undefined}
            >
              <span
                className="Select-multi-value-wrapper"
                id="react-select-3--value"
              >
                <div
                  className="Select-placeholder"
                >
                  Select Font Variants
                </div>
                <div
                  aria-activedescendant="react-select-3--value"
                  aria-disabled="true"
                  aria-expanded={false}
                  aria-label={undefined}
                  aria-labelledby={undefined}
                  aria-owns=""
                  className="Select-input"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  role="combobox"
                  style={
                    Object {
                      "border": 0,
                      "display": "inline-block",
                      "width": 1,
                    }
                  }
                  tabIndex={0}
                />
              </span>
              <span
                className="Select-arrow-zone"
                onMouseDown={[Function]}
              >
                <span
                  className="Select-arrow"
                  onMouseDown={[Function]}
                />
              </span>
            </div>
          </div>
          <div
            className="OtherProperties"
          >
            <div
              className="FontSize__Option OtherProperties__Option"
            >
              <label>
                Font size
              </label>
              <input
                max="248"
                min="9"
                name="quantity"
                onChange={[Function]}
                type="number"
                value={40}
              />
            </div>
            <div
              className="ColorOption__ForeGround OtherProperties__Option"
            >
              <label>
                Text Color
              </label>
              <div
                className="ColorPicker"
              >
                <div
                  className="ColorPicker__Swatch"
                  onClick={[Function]}
                >
                  <div
                    className="ColorPicker__Color"
                    style={
                      Object {
                        "color": "rgba(0, 0, 0, 1)",
                      }
                    }
                  >
                    A
                  </div>
                </div>
              </div>
            </div>
            <div
              className="ColorOption__BackGround OtherProperties__Option"
            >
              <label>
                Background Color
              </label>
              <div
                className="ColorPicker"
              >
                <div
                  className="ColorPicker__Swatch"
                  onClick={[Function]}
                >
                  <div
                    className="ColorPicker__Color"
                    style={
                      Object {
                        "backgroundColor": "rgba(255, 255, 255, 1)",
                      }
                    }
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
    </section>
    <div
      className="TexBoxContainer"
    >
      <div
        className="TextBox"
      >
        <div
          className="TextBox__TextArea"
        >
          <ul
            className="Texbx__FontDetails"
          >
            <li
              className="FontsDetails__Item FontsDetails__Item--FontFamily"
            >
              Oswald
            </li>
            <li
              className="FontsDetails__Item FontsDetails__Item--FontVariant"
            >
              Regular
            </li>
            <li
              className="FontsDetails__Item FontsDetails__Item--FontSize"
            >
              40
              px
            </li>
          </ul>
          <textarea
            autoFocus={true}
            placeholder="Write Something ..."
            style={
              Object {
                "color": "rgba(0, 0, 0, 1)",
                "fontFamily": "'Oswald', sans-serif",
                "fontSize": "40px",
                "fontWeight": 400,
              }
            }
          />
        </div>
        <div
          className="TextBox__CTA"
        >
          <button
            onClick={[Function]}
            type="button"
          >
            <i
              aria-hidden="true"
              className="fa fa-code"
            />
            How to use
          </button>
        </div>
      </div>
    </div>
  </section>
</div>
`;
